File: sandbox-images/cpp/harness/snake.cpp
#include "snake.hpp"

Snake::Snake() = default;
void Snake::set_id(int id)                  { id_ = id; }
void Snake::set_start_length(int)           {}
void Snake::set_start_position(Coord)       {}
void Snake::set_init_data(EnvInitData data) { init_data_ = std::move(data); }
std::pair<int, int> Snake::update(EnvStepData /*data*/) { return {0, 0}; }

SnakeBase* SnakeBase::create() { return new Snake(); }

File: sandbox-images/typescript/harness/usercode/snake.ts
import type { Coord, EnvInitData, EnvStepData } from '../types';

export class Snake {
  setId(_id: number): void {}
  setStartLength(_length: number): void {}
  setStartPosition(_pos: Coord): void {}
  setInitData(_data: EnvInitData): void {}
  update(_data: EnvStepData): [number, number] { return [0, 0]; }
}
